frontend/src/components/TaskItem/index.tsx

```diff
@@ -96,8 +98,23 @@ export default function TaskListItem(props: GetTaskFormProps) {
           </Grid>
         </ListItemText>
       </ListItemButton>
+
+      <EditTaskDialog 
+        open={openEdit}
+        task={props.task}
+        onClose={() => setOpenEdit(false)}
+        onSubmit={(formData) => {
+          console.log(formData);
+          setOpenEdit(false);
+          props.task.name = formData.name;
+          props.task.description = formData.description;
+          props.task.weight = formData.weight;
+          TaskService.updateTask(props.task, props.task.id!);
```

Comment:
why not just `TaskService.updateTask(formdata)`?

Reply:
Excellent point. Fixed.